Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -699,6 +699,36 @@ void RenderGrid::computeIntrinsicLogicalWidths(LayoutUnit& minLogicalWidth, Layo
 
     computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForColumns, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
 
+    if (isMasonry(GridTrackSizingDirection::ForColumns)) {
+        // The track sizing algorithm will only be run once in this case, since track sizing will not run in the masonry direction.
+        computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForRows, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
+
+        auto gridAxisTracksBeforeAutoPlacement = currentGrid().numTracks(GridTrackSizingDirection::ForRows);
+
+        for (auto* gridItem = grid.orderIterator().first(); gridItem; gridItem = grid.orderIterator().next()) {
+            if (grid.orderIterator().shouldSkipChild(*gridItem))
+                continue;
+
+            gridItem->setOverridingLogicalWidth(gridItem->minPreferredLogicalWidth());
+        }
+
+        m_masonryLayout.performMasonryPlacement(algorithm, gridAxisTracksBeforeAutoPlacement, GridTrackSizingDirection::ForColumns);
+
+
+        minLogicalWidth = m_masonryLayout.gridContentSize();
+
+        for (auto* gridItem = grid.orderIterator().first(); gridItem; gridItem = grid.orderIterator().next()) {
+            if (grid.orderIterator().shouldSkipChild(*gridItem))
+                continue;
+
+            gridItem->setOverridingLogicalWidth(gridItem->maxPreferredLogicalWidth());
```

Comment:
This code overrides the logical width of grid item's in all cases which may not be what we want? If the item has a specified logical width we probably want to keep that. I'm assuming we would only want to set the grid item's logical width to these values in certain scenarios.

Reply:
As discussed, added the necessary if flags.